netty/src/test/java/io/grpc/netty/ProtocolNegotiatorsTest.java

```diff
@@ -319,60 +446,65 @@ public void httpProxy_completes() throws Exception {
     LocalAddress proxy = new LocalAddress("httpProxy_completes");
     SocketAddress host = InetSocketAddress.createUnresolved("specialHost", 314);
 
-    ChannelInboundHandler mockHandler = mock(ChannelInboundHandler.class);
+    final BlockingQueue<Object> serverReads = new LinkedBlockingQueue<>();
```

Comment:
Why is the server code changing? This is effectively being rewritten and it isn't clear to my why.

Reply:
The server code is pretty unusual.   Using mocks to pull out the ctx and read the strings is surprising.  Also, by making the netty handler capture the HttpObject types, the connect request can be checked more precisely (rather than prefix matching the strings).  

I think the new version is much clearer, and more netty-like.  If you still feel it should be reverted I can.

Reply:
I don't really mind the blocking queue to get reads, although it is still unclear why this was changed. What is the behavior change this test was updated for?

I did prefer not to use the Http decoder stuff because it is less obvious we are using them correctly. Like we could receive the headers, but the request wasn't terminated. And for the encoder we'd need to make sure that there are no poor interactions with the CONNECT response.

Reply:
Originally this was failing due to grpchandler, and to check my bases I removed the mocks after getting some spurious channelInactives.  Later I added in the FakeGrpcHttp2Handler.  I think this is the root of the problem.  

I reverted these changes and added back in the mocks.